src/main/scala/mesosphere/marathon/core/task/termination/impl/TaskKillProgressActor.scala

```diff
@@ -0,0 +1,70 @@
+package mesosphere.marathon.core.task.termination.impl
+
+import akka.actor.{ Actor, ActorLogging, Props }
+import mesosphere.marathon.KillingTasksFailedException
+import mesosphere.marathon.core.task.Task
+import mesosphere.marathon.core.event.MesosStatusUpdateEvent
+
+import scala.concurrent.Promise
+import scala.collection.mutable
+import scala.util.Try
+
+/**
+  * This actor watches over a given set of taskIds and completes a promise when
+  * all tasks have been reported terminal (including LOST et al)
```

Comment:
Why/since when is `MesosStatusUpdateEvent` deprecated? `mesosphere.marathon.core.task.bus` is, somehow, but that doesn't include publishing `MesosStatusUpdateEvent` on the eventStream afaik.

The point of this actor is: it waits until all the tasks it watches are terminal and then completes a promise.

Before I introduced this, there were 2 Actors (TaskKillActor, AppStopActor) that mixed in a common behavior that killed tasks and waited until they'd be reported terminal. The whole concern of killing/throttling/retrying is moved into the TaskKillServiceActor, and the watch dogs are here only to allow waiting for a future when all requested task kills have happened.